From e8968fd3f120e76d9eb4268385747d256c9ff19b Mon Sep 17 00:00:00 2001
From: Ches Martin <ches@whiskeyandgrits.net>
Date: Fri, 3 Dec 2010 21:44:08 +0700
Subject: [PATCH] Include script.aculo.us with --javascript app generator option

The support for --javascript and --skip_javascript options added in
708e09448ba41c45f189c6b1adc989eba8994b20 checks for the wrong option when
determining whether or not to copy the script.aculo.us JS files.
---
 .../rails/generators/rails/app/app_generator.rb    |    8 ++++----
 railties/test/generators/app_generator_test.rb     |    6 ++++++
 2 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/railties/lib/rails/generators/rails/app/app_generator.rb b/railties/lib/rails/generators/rails/app/app_generator.rb
index ef1eb8d..f86200b 100644
--- a/railties/lib/rails/generators/rails/app/app_generator.rb
+++ b/railties/lib/rails/generators/rails/app/app_generator.rb
@@ -104,18 +104,18 @@ module Rails
 
     def javascripts
       empty_directory "public/javascripts"
-          
+
       unless options[:skip_javascript]
         copy_file "public/javascripts/#{@options[:javascript]}.js"
         copy_file "public/javascripts/#{@options[:javascript]}_ujs.js", "public/javascripts/rails.js"
-        
-        if options[:prototype]
+
+        if options[:javascript] == "prototype"
           copy_file "public/javascripts/controls.js"
           copy_file "public/javascripts/dragdrop.js"
           copy_file "public/javascripts/effects.js"
         end
       end
-      
+
       copy_file "public/javascripts/application.js"
     end
 
diff --git a/railties/test/generators/app_generator_test.rb b/railties/test/generators/app_generator_test.rb
index 7faa674..5d0ebb9 100644
--- a/railties/test/generators/app_generator_test.rb
+++ b/railties/test/generators/app_generator_test.rb
@@ -134,6 +134,9 @@ class AppGeneratorTest < Rails::Generators::TestCase
     assert_file "config/application.rb", /#\s+config\.action_view\.javascript_expansions\[:defaults\]\s+=\s+%w\(jquery rails\)/
     assert_file "public/javascripts/application.js"
     assert_file "public/javascripts/prototype.js"
+    assert_file "public/javascripts/controls.js"
+    assert_file "public/javascripts/dragdrop.js"
+    assert_file "public/javascripts/effects.js"
     assert_file "public/javascripts/rails.js"
     assert_file "test"
   end
@@ -151,6 +154,9 @@ class AppGeneratorTest < Rails::Generators::TestCase
     assert_file "config/application.rb", /#\s+config\.action_view\.javascript_expansions\[:defaults\]\s+=\s+%w\(jquery rails\)/
     assert_file "public/javascripts/application.js"
     assert_file "public/javascripts/prototype.js"
+    assert_file "public/javascripts/controls.js"
+    assert_file "public/javascripts/dragdrop.js"
+    assert_file "public/javascripts/effects.js"
     assert_file "public/javascripts/rails.js", /prototype/
   end
 
-- 
1.7.1

